test/unit/core/adapters/personalization.ts

```diff
@@ -0,0 +1,380 @@
+
+      expect(refinement).to.eql(payload);
+    });
+
+    it('should return undefined if Action type is incorrect', () => {
+      const type = 'aa';
+      const action: any = { type, payload: {} };
+      const refinementCrumb = stub(Selectors, 'refinementCrumb');
+
+      const refinement = Adapter.extractRefinement(action, state);
+
+      expect(refinement).to.eql({ value: undefined, field: undefined });
+      expect(refinementCrumb).to.not.have.been.called;
+    });
+  });
+
+  describe('generateNewBias()', () => {
+    it('should generate new bias', () => {
+      stub(Date, 'now').returns(1000000);
+
+      const result = Adapter.generateNewBias();
+
+      expect(result).to.deep.equal({ lastUsed: 1000000 / 1000 });
+    });
+  });
+
+  describe('transformToBrowser()', () => {
+    it('should return empty array if the state does not have allIds', () => {
+      const state: any = { expiry: 2000 };
+      const key = 'test';
+
+      const result = Adapter.transformToBrowser(state, key);
+
+      expect(result).to.deep.equal({
+        allIds: []
+      });
+    });
+
+    it('should return data in certain format when allIds exist in the state', () => {
+      const state: any = {
+        allIds: [{
+          variant: 'color',
+          key: 'blue',
+        }],
+        byId: {
+          color: {
+            blue: {
+              lastUsed: 11111
+            }
+          }
+        }
+      };
+      const key = 'test';
+
+      const result = Adapter.transformToBrowser(state, key);
+
+      expect(result).to.deep.equal({
+        allIds: [{
+          variant: 'color',
+          key: 'blue',
+          lastUsed: 11111
+        }]
+      });
+    });
+  });
+
+  describe('transformFromBrowser()', () => {
+    it('should filter allIds that are older than the set time', () => {
+      const oneDayInSec = 86400;
+      const now = Date.now() / 1000;
+      const expiry = 30;
+      const browserStorage = {
+        allIds: [{ variant: 'color', key: 'blue', lastUsed: now - oneDayInSec * 5 },
+                 { variant: 'brand', key: 'Shoe', lastUsed: now - oneDayInSec * 6 },
+                 { variant: 'color', key: 'red', lastUsed: now - oneDayInSec * 21 },
+                 { variant: 'brand', key: 'Nike', lastUsed: now - oneDayInSec * 33 }]
+      };
+      const biasFromBrowser = {
+        allIds: [{ variant: 'color', key: 'blue' }, { variant: 'brand', key: 'Shoe'},
+                 { variant: 'color', key: 'red' }],
+        byId: {
+          brand: {
+            Shoe: {
+              lastUsed: now - oneDayInSec * 6
+            }
+          },
+          color: {
+            blue: {
+              lastUsed: now - oneDayInSec * 5
+            },
+            red: {
+              lastUsed: now - oneDayInSec * 21
+            }
+          }
+        }
+      };
+      const realTimeBiasing = 'real';
+      const conf = { personalization: { realTimeBiasing } };
+      const state: any = 's';
```

Comment:
if we're going to make a mock state, please at least make it the correct type: `{}`